jax/numpy/lax_numpy.py

```diff
@@ -1135,7 +1135,7 @@ def check(size):
       int, size, "The error arose in jax.numpy.reshape.")
   newshape = [check(size) for size in newshape] if iterable else check(newshape)
   newsize = _prod((newshape,) if type(newshape) is Poly else newshape)
-  if newsize < 0:
+  if np.any(np.equal(newshape, -1)):
```

Comment:
I think this is a better check than before. I am wondering though if other negative dimensions will be caught reliably somewhere else.

Reply:
Comment a few lines above said: "# other errors, like having more than one -1, are caught downstream".
In any case, the change does not worsen error handling.